### DISCUSSION TITLE
Workbench Templates

### DISCUSSION THREAD

**Discussion by oursland** [FreeCAD/FreeCAD#18254] [Code contribution discussions]
There are different means for generating a workbench from a template, but it is unclear which is the recommended approach.

* [FreeCAD/freecad.workbench_starterkit](https://github.com/FreeCAD/freecad.workbench_starterkit)
* [fcbt/CreateModule.py](https://github.com/FreeCAD/FreeCAD/blob/main/src/Tools/fcbt/CreateModule.py)
* [fcbt/CreatePyModule.py](https://github.com/FreeCAD/FreeCAD/blob/main/src/Tools/fcbt/CreatePyModule.py)

@wwmayer It appears that the last substantial change to `fcbt` was in 2021.  Are the `fcbt` paths still current?

**Comment by wwmayer** [comment]
> Are the fcbt paths still current?

Current?

**Reply by oursland** [reply]
Current as in recommended for use.

**Reply by wwmayer** [reply]
Well, it still works as expected. I have created a test wb in C++ and Python.
The difference to the start kit is that this is mostly aimed for the creation of add-ons because it's Python-only and creates some extra files.
